Gems/ROS2/Code/Source/Lidar/ROS2Lidar2DSensorComponent.cpp

```diff
@@ -0,0 +1,295 @@
+                        "Lidar Implementation",
+                        "Select a lidar implementation out of registered ones.")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &ROS2Lidar2DSensorComponent::OnLidarImplementationSelected)
+                    ->Attribute(AZ::Edit::Attributes::StringList, &ROS2Lidar2DSensorComponent::FetchLidarSystemList)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::EntityId,
+                        &ROS2Lidar2DSensorComponent::m_lidarParameters,
+                        "Lidar parameters",
+                        "Configuration of Custom lidar")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &ROS2Lidar2DSensorComponent::IsConfigurationVisible)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::ComboBox,
+                        &ROS2Lidar2DSensorComponent::m_ignoreLayer,
+                        "Ignore layer",
+                        "Should we ignore selected layer index")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &ROS2Lidar2DSensorComponent::IsIgnoredLayerConfigurationVisible)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ROS2Lidar2DSensorComponent::m_ignoredLayerIndex,
+                        "Ignored layer index",
+                        "Layer index to ignore")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &ROS2Lidar2DSensorComponent::IsIgnoredLayerConfigurationVisible)
+                    ->DataElement(
+                        0, &ROS2Lidar2DSensorComponent::m_excludedEntities, "Excluded Entities", "List of entities excluded from raycasting.")
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->Attribute(AZ::Edit::Attributes::ContainerCanBeModified, true)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &ROS2Lidar2DSensorComponent::IsEntityExclusionVisible)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ROS2Lidar2DSensorComponent::m_addPointsAtMax,
+                        "Points at Max",
+                        "If set true LiDAR will produce points at max range for free space")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, &ROS2Lidar2DSensorComponent::IsMaxPointsConfigurationVisible);
+            }
+        }
+    }
+    static AZStd::string GetDefaultLidarSystem()
+    {
+        const auto& lidarSystemList = LidarRegistrarInterface::Get()->GetRegisteredLidarSystems();
+        if (lidarSystemList.empty())
+        {
+            AZ_Warning("ROS2LidarSensorComponent", false, "No LIDAR system for the sensor to use.");
+            return {};
+        }
+        return lidarSystemList.front();
+    }
+
+    void ROS2Lidar2DSensorComponent::FetchLidarImplementationFeatures()
```

Comment:
I would recommend considering:
 - using only one implementation (Scene Query)
 - or test against RGL.
 @adamdbrw do you think that supporting RGL would be beneficial for 2D lasers or not here?
 RGL would use visual meshes and can support hundreds of those 2D lidars.

Reply:
We agreed to allow multiple implementations. Tested with RGL and found compilation issue addressed in one of next commits